flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/generated/BoundComparator.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.generated;
+
+import org.apache.flink.table.dataformat.BaseRow;
+
+import java.io.Serializable;
+
+/**
+ * comparing boundary values.
+ */
+public interface BoundComparator extends Serializable {
+
+	/**
+	 * reset the comparator.
+	 */
+	void reset();
```

Comment:
why a comparator need to be reset?

Reply:
Because the field getter of currentRow may be cached.
For example, the `UnboundedFollowingOverWindowFrame`, when calculating its tail boundary, may cross multiple Rows in iterator. The `BoundComparator.compare` will be invoked many times is the same `currentRow` input, so it is not necessary to get its field value every time, so cached its field value in `BoundComparator`.
`reset` is designed to clear the cache of `BoundComparator`.